packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -34,23 +35,95 @@ export const CONTROL_PANEL_CLASS = 'theia-ControlPanel';
 
 @injectable()
 export class FileDialogProps extends DialogProps {
+    /**
+     * A human-readable string for the open button.
+     */
+    openLabel?: string;
+
+    /**
+     * Allow to select files, defaults to `true`.
+     */
+    canSelectFiles?: boolean;
+
+    /**
+     * Allow to select folders, defaults to `false`.
+     */
+    canSelectFolders?: boolean;
+
+    /**
+     * Allow to select many files or folders.
+     */
+    canSelectMany?: boolean;
+
+    /**
+     * A set of file filters that are used by the dialog. Each entry is a human readable label,
+     * like "TypeScript", and an array of extensions, e.g.
+     * ```ts
+     * {
+     * 	'Images': ['png', 'jpg']
+     * 	'TypeScript': ['ts', 'tsx']
+     * }
+     * ```
+     */
+    filters?: FileDialogTreeFilters;
+}
+
+export type FileSelection = Readonly<FileStatNode> | Readonly<FileStatNode>[] | undefined;
+
+export namespace FileSelection {
```

Comment:
Sorry for the confusion, I meant you can just use `UriSelection`, i.e.
```
const uri = UriSelection.getUri(await dialog.open());
const URIs = UriSelection.getUris(await dialog.open());
```

Reply:
Well. How can we use URI here?
https://github.com/theia-ide/theia/blob/68e8ef46e5aab9cc82b7546d565704c9ae86035a/packages/workspace/src/browser/workspace-frontend-contribution.ts#L80

Reply:
Why do we expect multiple fils here? It could be:
```ts
if (FileStatNode.is(node)) {
   this.open(node);
} 
```

Reply:
It was a workaround. But it could be working when the dialog will return an array.
I used `MaybeArray` for the file dialog
```
class FileDialog extends AbstractDialog<MaybeArray<FileStatNode>>
```
Then it gives `FileStatNode | FileStatNode[] | undeifned`
But `FileStatNode.is(..)` receives only `node: TreeNode | undefined` and it's not suitable for an array. I think we have to improve `FileStatNode.is(..)` somehow.

Reply:
feel free to change `FileStatNode.is` that it accepts an arg of `object | undefined` type: https://github.com/theia-ide/theia/blob/3ef537c96f57cb1c9ebb1b331e8c93bc0fb74db1/packages/filesystem/src/browser/file-tree/file-tree.ts#L96

Reply:
I like it :)